libmscore/synthesizerstate.cpp

```diff
@@ -54,7 +55,15 @@ void SynthesizerState::read(XmlReader& e)
                   else
                         e.unknown();
                   }
-            push_back(group);
+            tempGroups.push_back(group);
+            }
+
+      if (int(tempGroups.size()) > 0) {
+            // Remove any previously set state if we have read a new state
+            clear();
+            for (auto& g : tempGroups)
+                  push_back(g);
```

Comment:
Wouldn't something like `swap(tempGroups)` work? It would probably be faster as there will be no need in copying elements.

Reply:
Sure, I didn't know that was a feature. I'll change it to that.